From 749e9d221960c6703220eaf4c99b6ee913db7607 Mon Sep 17 00:00:00 2001
From: Christopher Kohlhoff <chris@kohlhoff.com>
Date: Tue, 17 Sep 2013 21:43:29 +0000
Subject: [PATCH] Ensure signal number is correctly passed to the completion
 handler when starting an async_wait on a signal that is already raised.

[SVN r85746]
---
 include/boost/asio/detail/impl/signal_set_service.ipp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/boost/asio/detail/impl/signal_set_service.ipp b/include/boost/asio/detail/impl/signal_set_service.ipp
index 131b5af0..bed517b5 100644
--- a/include/boost/asio/detail/impl/signal_set_service.ipp
+++ b/include/boost/asio/detail/impl/signal_set_service.ipp
@@ -589,6 +589,7 @@ void signal_set_service::start_wait_op(
     if (reg->undelivered_ > 0)
     {
       --reg->undelivered_;
+      op->signal_number_ = reg->signal_number_;
       io_service_.post_deferred_completion(op);
       return;
     }
